presto-main/src/main/java/com/facebook/presto/operator/aggregation/partial/SkipAggregationBuilder.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.partial;
+
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.block.LongArrayBlock;
+import com.facebook.presto.memory.context.LocalMemoryContext;
+import com.facebook.presto.operator.CompletedWork;
+import com.facebook.presto.operator.HashCollisionsCounter;
+import com.facebook.presto.operator.UpdateMemory;
+import com.facebook.presto.operator.Work;
+import com.facebook.presto.operator.WorkProcessor;
+import com.facebook.presto.operator.aggregation.AccumulatorFactory;
+import com.facebook.presto.operator.aggregation.builder.HashAggregationBuilder;
+import com.facebook.presto.spi.function.aggregation.GroupByIdBlock;
+import com.facebook.presto.spi.function.aggregation.GroupedAccumulator;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.common.Utils.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * {@link HashAggregationBuilder} that does not aggregate input rows at all.
+ * It passes the input pages, augmented with initial accumulator state to the output.
+ * It can only be used at the partial aggregation step as it relies on rows be aggregated at the final step.
+ */
+public class SkipAggregationBuilder
+        implements HashAggregationBuilder
+{
+    private final LocalMemoryContext memoryContext;
+    private final List<GroupedAccumulator> groupedAccumulators;
+    @Nullable
+    private Page currentPage;
+    private final int[] hashChannels;
+
+    public SkipAggregationBuilder(
+            List<Integer> groupByChannels,
+            Optional<Integer> inputHashChannel,
+            List<AccumulatorFactory> accumulatorFactories,
+            LocalMemoryContext memoryContext)
+    {
+        this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
+        this.groupedAccumulators = accumulatorFactories.stream()
```

Comment:
add `requireNonNull(accumulatorFactories, "accumulatorFactories is null");` before this line

Reply:
done